### DISCUSSION TITLE
Is the type of the builtin var retval an int64 or an uint64?

### DISCUSSION THREAD

**Discussion by Phoenix500526** [bpftrace/bpftrace#3672] [Q&A] [answered]
In the bpftrace [manual](https://github.com/bpftrace/bpftrace/blob/master/man/adoc/bpftrace.adoc#builtins), the type of the built-in variable `retval` is int64. However, in the following bpftrace one-liner script: 

```bash
sudo bpftrace -e 'kretprobe:check_mem_size_reg /retval==-13/ { printf("retval: %d\n", retval);print(kstack());}'
```

, an error message is shown:

```bash
stdin:1:30-39: WARNING: comparison of integers of different signs: 'unsigned int64' and 'int64' can lead to undefined behavior
```

So, is the type of retval `unsigned int64` or `int64`? Is it a bug? Or just a typo in the manual?

**Comment by jordalgo** [comment, marked as answer]
@Phoenix500526 For kret and uret probes, it looks like a uint64 - https://github.com/bpftrace/bpftrace/blob/master/src/ast/passes/semantic_analyser.cpp#L418

But for fentry/fexit it looks like it depends and can be anything.

**Reply by Phoenix500526** [reply]
Shall we correct the doc? I'm happy to issue a pr to correct the doc.

**Reply by jordalgo** [reply]
@Phoenix500526 Please do! PRs welcome.
